test/optimizer/plugins/AddPathRulesOptimizer.test.ts

```diff
@@ -263,5 +372,251 @@ describe('optimizer', () => {
+      optimizer.optimize(myPackage);
+
+      expect(profile.rules).toEqual([categoryOnlyCC, categorySlicingOrdered, categorySlicingRules]);
+    });
+
+    it('should handle paths that require multiple rules at the same path', () => {
+      // Profile: MultipleRulesPerPathObs
+      // Parent: Observation
+      // * category 1..1 MS
+      // * category only CodeableConcept
+      // * category.coding 1..* MS
+      // * category.coding only Coding
+      // * category.coding.system 1..1 MS
+      // * category.coding.system only uri
+      const profile = new ExportableProfile('MultipleRulesPerPathObs');
+      profile.parent = 'Observation';
+      const categoryCard = new ExportableCardRule('category');
+      categoryCard.min = 1;
+      categoryCard.max = '1';
+      const categoryMS = new ExportableFlagRule('category');
+      categoryMS.mustSupport = true;
+      const categoryOnlyCC = new ExportableOnlyRule('category');
+      categoryOnlyCC.types = [{ type: 'CodeableConcept' }];
+
+      const categoryCodingCard = new ExportableCardRule('category.coding');
+      categoryCodingCard.min = 1;
+      categoryCodingCard.max = '*';
+      const categoryCodingMS = new ExportableFlagRule('category.coding');
+      categoryCodingMS.mustSupport = true;
+      const categoryCodingOnlyCC = new ExportableOnlyRule('category.coding');
+      categoryCodingOnlyCC.types = [{ type: 'Coding' }];
+
+      const categoryCodingSystemCard = new ExportableCardRule('category.coding.system');
+      categoryCodingSystemCard.min = 1;
+      categoryCodingSystemCard.max = '1';
+      const categoryCodingSystemMS = new ExportableFlagRule('category.coding.system');
+      categoryCodingSystemMS.mustSupport = true;
+      const categoryCodingSystemOnlyCC = new ExportableOnlyRule('category.coding.system');
+      categoryCodingSystemOnlyCC.types = [{ type: 'uri' }];
+
+      profile.rules.push(
+        categoryCard,
+        categoryMS,
+        categoryOnlyCC,
+        categoryCodingCard,
+        categoryCodingMS,
+        categoryCodingOnlyCC,
+        categoryCodingSystemCard,
+        categoryCodingSystemMS,
+        categoryCodingSystemOnlyCC
+      );
+
+      myPackage.add(profile);
+      optimizer.optimize(myPackage);
+
+      expect(profile.rules).toEqual([
+        categoryCard,
+        categoryMS,
+        categoryOnlyCC,
+        categoryCodingCard,
+        categoryCodingMS,
+        categoryCodingOnlyCC,
+        categoryCodingSystemCard,
+        categoryCodingSystemMS,
+        categoryCodingSystemOnlyCC
+      ]);
+    });
+
+    it('should optimize order for rules that require multiple rules at the same path', () => {
+      // Profile: MultipleRulesPerPathObs
+      // Parent: Observation
+      // * category 1..1 MS
+      // * category only CodeableConcept
+      // * category.coding 1..* MS
+      // * category.coding only Coding
+      // * category.coding.system 1..1 MS
+      // * category.coding.system only uri
```

Comment:
In this test, you put the `only` rules at the end, so I think this FSH should reflect that.

Reply:
Ah good catch. I had it in the expected order. I've updated the comment, but I realized I didn't have a comment above the expect statements that has the expected FSH. Do you think this would be helpful to include? I was thinking the expect statements were enough, but I realized most other tests in this file do have the second commented FSH block.